File: state/c-sharp/VendingMachine.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace State
{
    // Durumu izlenecek olan sınıftır.
    // UML diyagramındaki Context yapısına denk gelmektedir.
    public class VendingMachine
    {
        private VendingMachineState _state;

        // Otomata örneği oluştuğu zaman başlangıç durumuna geçiyor.
        public VendingMachine()
        {
            SetState(new InitializeState());
        }

        public void SetState(VendingMachineState state)
        {
            _state = state;
            _state.HandleState(this);
        }

        public void RequestProduct(string productName, double price)
        {
            // Burada ürünlere ait iş kodları olacaktır.
            // Ürün kontrolü, fiyatlandırma gibi kontroller.
            // Ürün ve fiyat olarak bir problemin olmadığı var sayılsın.

            SetState(new PreparingState());
        }
    }
}

File: state/go/vending-machine.go
package main

// VendingMachine Durumu izlenecek olan sınıftır.
// UML diyagramındaki Context yapısına denk gelmektedir.
type VendingMachine struct {
	state VendingMachineState
}

// NewVendingMachine Otomat örneği oluştuğu zaman başlangıç durumuna geçiyor.
func NewVendingMachine() *VendingMachine {
	vm := &VendingMachine{}
	vm.SetState(NewInitializeState())
	return vm
}

func (m *VendingMachine) SetState(state VendingMachineState) {
	m.state = state
	m.state.HandleState(m)
}

func (m *VendingMachine) RequestProduct(_ string, _ int) {
	// Burada ürünlere ait iş kodları olacaktır.
	// Ürün kontrolü, fiyatlandırma gibi kontroller.
	// Ürün ve fiyat olarak bir problemin olmadığı var sayılsın.

	m.SetState(NewPreparingState())
}
